src/FMBot.Bot/Commands/FriendsCommands.cs

```diff
@@ -173,65 +173,52 @@ public async Task AddFriends([Summary("Friend names")] params string[] friends)
-                                {
-                                    await this._friendsService.AddLastFMFriendAsync(this.Context.User.Id,
-                                        friend);
-                                    addedFriendsCount++;
-                                }
-                                else
-                                {
-                                    friendNotFoundList.Add(friend);
-                                }
-                            }
-                            else
-                            {
-                                await this._friendsService.AddDiscordFriendAsync(this.Context.User.Id, friendUser.Id);
-                                addedFriendsCount++;
-                            }
-                        }
-                        else
+                    Persistence.Domain.Models.User friendUserSettings;
+                    string username;
+
+                    if (friendGuildUser != null) {
+                        friendUserSettings = await this._userService.GetUserSettingsAsync(friendGuildUser);
+                        username = friendUserSettings?.UserNameLastFM ?? friend;
+                    } else {
+                        username = friend;
+                        friendUserSettings = null;
+                    }
+
+                    if (!existingFriends.Select(s => s.ToLower()).Contains(username))
+                    {
+                        if (await this._lastFmService.LastFMUserExistsAsync(username))
                         {
-                            if (await this._lastFmService.LastFMUserExistsAsync(friend))
-                            {
-                                await this._friendsService.AddLastFMFriendAsync(this.Context.User.Id,
-                                    friend);
-                                addedFriendsCount++;
-                            }
-                            else
-                            {
-                                friendNotFoundList.Add(friend);
-                            }
+                            await this._friendsService.AddLastFMFriendAsync(this.Context.User.Id, username, friendUserSettings?.UserId);
+                            addedFriendsCount++;
+                        } else {
+                            friendNotFoundList.Add(username);
                         }
+                    } else {
+                        duplicateFriendsList.Add(username);
                     }
                 }
 
                 if (addedFriendsCount > 1)
                 {
-                    await ReplyAsync("Succesfully added " + addedFriendsCount + " friends.");
+                    await ReplyAsync("Succesfully added " + addedFriendsCount + " friend" + 
+                        (addedFriendsCount == 1 ? "s." : ".")
+                    );
                 }
                 else if (addedFriendsCount == 1)
```

Comment:
This can be removed, you already fix the reply for 1 added friend a few lines above this.